Block/SurchargeTotal.php

```diff
@@ -0,0 +1,41 @@
+<?php
+/**
+ * @by SwiftOtter, Inc. 4/19/17
+ * @website https://swiftotter.com
+ **/
+
+namespace SwiftOtter\ShippingSurcharge\Block;
+
+use \SwiftOtter\ShippingSurcharge\Model\Surcharge as SurchargeModel;
+
+trait SurchargeTotal
+{
+    /**
+     * @var \SwiftOtter\ShippingSurcharge\Config\Info
+     */
+    private $configInfo;
+
+    public function __construct(\SwiftOtter\ShippingSurcharge\Config\Info $configInfo)
+    {
+        $this->configInfo = $configInfo;
+    }
+
+    private function initSurcharge()
+    {
+        if (!$this->configInfo->isFeatureEnabled() || !$this->getSource()) return;
+
+        /** @var $parent \Magento\Sales\Block\Adminhtml\Order\Invoice\Totals */
+        $parent = $this->getParentBlock();
+
+        $surcharge = new \Magento\Framework\DataObject(
```

Comment:
It might be a good idea to alias some of these class names as we're adding on to this. I take responsibility for not doing that in the first place.